lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_lint_localizations.rb

```diff
@@ -22,17 +32,38 @@ def self.run_linter(params)
           install_path: resolve_path(params[:install_path]),
           version: params[:version]
         )
-        all_violations = helper.run(
+
+        helper.run(
           input_dir: resolve_path(params[:input_dir]),
           base_lang: params[:base_lang],
           only_langs: params[:only_langs]
         )
+      end
+
+      def self.report(violations:, base_lang:)
+        violations.each do |lang, lang_violations|
+          UI.error "Inconsistencies found between '#{base_lang}' and '#{lang}':\n\n#{lang_violations.join("\n")}\n"
+        end
+      end
+
+      def self.find_duplicated_keys(params)
+        duplicate_keys = {}
 
-        all_violations.each do |lang, lang_violations|
-          UI.error "Inconsistencies found between '#{params[:base_lang]}' and '#{lang}':\n\n#{lang_violations.join("\n")}\n"
+        files_to_lint = Dir.chdir(params[:input_dir]) do
+          Dir.glob('*.lproj/Localizable.strings').map do |file|
+            {
+              language: File.basename(File.dirname(file), '.lproj'),
+              path: File.join(params[:input_dir], file)
+            }
+          end
         end
 
-        all_violations
+        files_to_lint.each do |file|
+          duplicates = Fastlane::Helper::Ios::StringsFileValidationHelper.find_duplicated_keys(file: file[:path])
+          duplicate_keys[file[:language]] = duplicates.map { |key, value| "`#{key}` was found at multiple lines: #{value.join(', ')}" } unless duplicates.empty?
+        end
```

Comment:
If you want to get fancy, you could even consider replacing the whole `duplicate_keys = {}`+`.each` + `[return] duplicate_keys` with a `map` + `to_h.compact`, making the whole thing even shorter:

```ruby
def self.find_duplicated_keys(params)
  Dir.glob('*.lproj/Localizable.strings', base: params[:input_dir]).map do |file|
    path = File.join(params[:input_dir], file)
    duplicates = Fastlane::Helper::Ios::StringsFileValidationHelper.find_duplicated_keys(file: path)
    language = File.basename(File.dirname(file), '.lproj')
    [language, duplicates.map { |key, value| "`#{key}` was found at multiple lines: #{value.join(', ')}" } unless duplicates.empty?]
  end.to_h.compact # remove `nil` values for languages with no duplicates
end
```

Alternatively, you can remove the `unless duplicates.empty?` in the loop, and filter out empty entries (instead of `nil` ones via `compact`) explicitly. I have no preference here, it mostly depend on what you prefer and find more readable (… except maybe that by being more explicit we could then consider avoiding the need for a `# comment` on what the `compact` was for?)

```ruby
def self.find_duplicated_keys(params)
  Dir.glob('*.lproj/Localizable.strings', base: params[:input_dir]).map do |file|
    path = File.join(params[:input_dir], file)
    duplicates = Fastlane::Helper::Ios::StringsFileValidationHelper.find_duplicated_keys(file: path)
    language = File.basename(File.dirname(file), '.lproj')
    [language, duplicates.map { |key, value| "`#{key}` was found at multiple lines: #{value.join(', ')}" }]
  end.to_h.reject { |_, v| v.empty? }
end
```